### DISCUSSION TITLE
Tweeting hosted images

### DISCUSSION THREAD

**Discussion by karakotineeraj** [plhery/node-twitter-api-v2#192] [Q&A] [answered]
Hi @PLhery @HapLifeMan @a2441918 @filippkowalski @mika-f, I was wondering if there is any way to **tweet an image that is hosted online** with this tool?
Thanking in advance

**Comment by alkihis** [comment, marked as answer]
Hi, 
Just download the image locally to a `Buffer` or file system, then upload it to Twitter.

**Reply by karakotineeraj** [reply]
Thanks but that means there is no option in the library itself?

**Reply by plhery** [reply]
I feel it's a common request, not adding a lot of code... Let's consider it if there are other upvotes @alkihis?

**Reply by karakotineeraj** [reply]
Can I create this feature? I would very much like to contribute to this repository.

**Reply by alkihis** [reply]
@PLhery I don't think this feature belongs to the lib. HTTP code is optimized for handling twitter responses, not raw responses of any domain.

Libs like axios or node-fetch handle Buffer responses better and maybe are more appropriate I think 🤷

**Reply by plhery** [reply]
Makes sense :) thanks for the proposition @karakotineeraj!

It's 2 lines for the dev who wants to do it, so I agree it's better if the dev tweaks/configures/debugs these two lines the way they want :)

**Reply by plhery** [reply]
Makes sense :) thanks for the proposition @karakotineeraj!

It's 2 lines for the dev who wants to do it, so I agree it's better if the dev tweaks/configures/debugs these two lines the way they want :)

**Reply by karakotineeraj** [reply]
> Makes sense :) thanks for the proposition @karakotineeraj!
> 
> It's 2 lines for the dev who wants to do it, so I agree it's better if the dev tweaks/configures/debugs these two lines the way they want :)

Yeah I think it would be better that way and also @alkihis has made a good point too.
